projects/plugins/jetpack/extensions/blocks/recurring-payments/edit.jsx

```diff
@@ -1,543 +1,68 @@
-						onClick={ () => this.setState( { addingMembershipAmount: PRODUCT_NOT_ADDING } ) }
-					>
-						{ __( 'Cancel', 'jetpack' ) }
-					</Button>
-				</div>
-			</div>
-		);
-	};
-	getFormattedPriceByProductId = id => {
-		const product = this.state.products
-			.filter( prod => parseInt( prod.id ) === parseInt( id ) )
-			.pop();
-		return formatCurrency( parseFloat( product.price ), product.currency );
-	};
-
-	setMembershipAmount = id => {
-		const { innerButtons, updateBlockAttributes, setAttributes } = this.props;
-		const currentPlanId = this.props.attributes.planId;
-		const defaultTextForNewPlan = sprintf(
-			/* translators: placeholder is an amount of money. */
-			__( '%s contribution', 'jetpack' ),
-			this.getFormattedPriceByProductId( id )
-		);
-		const defaultTextForCurrentPlan = currentPlanId
-			? sprintf(
-					/* translators: placeholder is an amount of money. */
-					__( '%s contribution', 'jetpack' ),
-					this.getFormattedPriceByProductId( currentPlanId )
-			  )
-			: undefined;
-
-		if ( innerButtons && innerButtons.length ) {
-			innerButtons[ 0 ].innerBlocks.forEach( block => {
-				const currentText = block.attributes.text;
-				const text =
-					currentText === defaultTextForCurrentPlan ? defaultTextForNewPlan : currentText;
-				updateBlockAttributes( block.clientId, { text } );
-			} );
-		}
+import ProductManagementControls from '../../shared/components/product-management-controls';
+
+export default function Edit( props ) {
+	const { setAttributes, attributes, clientId } = props;
+	const { planId } = attributes;
+	const postLink = useSelect(
+		select => new URL( select( 'core/editor' ).getCurrentPost().link ),
+		[]
+	);
 
-		const planId = parseInt( id );
-		this.props.postLink.searchParams.set( 'recurring_payments', planId.toString() );
+	const updateSubscriptionPlan = newPlanId => {
+		postLink.searchParams.set( 'recurring_payments', newPlanId );
 		return setAttributes( {
-			planId,
+			planId: newPlanId,
 			// We need to use an absolute URL otherwise the reader won't be able to open the post.
-			url: this.props.postLink.toString(),
-			uniqueId: `recurring-payments-${ planId }`,
+			url: postLink.toString(),
+			uniqueId: `recurring-payments-${ newPlanId }`,
 		} );
 	};
 
-	renderMembershipAmounts = () => (
-		<div>
-			{ this.state.products.map( product => (
-				<Button
-					className="membership-button__field-button"
-					variant="secondary"
-					key={ product.id }
-					onClick={ () => this.setMembershipAmount( product.id ) }
-				>
-					{ formatProductAmount( product ) }
-				</Button>
-			) ) }
-		</div>
-	);
-
-	renderDisclaimer = () => {
-		return (
-			<div className="membership-button__disclaimer">
-				<ExternalLink href="https://wordpress.com/support/wordpress-editor/blocks/payments/#related-fees">
-					{ __( 'Read more about Payments and related fees.', 'jetpack' ) }
-				</ExternalLink>
-			</div>
-		);
-	};
-
-	renderUpgradeNudges = () => {
```

Comment:
This feature will be re-implemented in a follow-up issue. This upgrade nudge only affects standalone WordPress sites that have the Jetpack plugin installed and are on a free Jetpack subscription.

The WPCOM upgrade nudge will still be shown as always.

Reply:
I'll merge your branch into this one.

Reply:
> The [availability checks](https://github.com/Automattic/jetpack/pull/23580/files#diff-a41de8578e653f89cd1ace91e14f13ed157f6c27cd01c4dfa5c251de6bf9bf03R53) are used to determine if the block is showing the WPCOM nudge. Kind of not ideal, huh.

These checks can be simplified: https://github.com/Automattic/jetpack/pull/23580/files#r834136333
Since I'll be merging your branch into this one I'll apply the simplification here.

Reply:
Merged.

Reply:
I'm going to add the link to fees again since it won't be shown when a user doesn't have a paid Jetpack plan.